vfio/src/vfio_pci.rs

```diff
@@ -676,6 +678,18 @@ impl VfioPciDevice {
 
 impl Drop for VfioPciDevice {
     fn drop(&mut self) {
+        for region in self.mmio_regions.iter() {
+            if let (Some(addr), Some(size)) = (region.host_addr, region.mmap_size) {
+                let ret = unsafe { libc::munmap(addr as *mut libc::c_void, size) };
+                if ret != 0 {
+                    error!(
+                        "Could not unmap regions, error:{}",
+                        io::Error::last_os_error()
+                    );
+                }
+            }
+        }
```

Comment:
Great! Thanks for fixing this.
Could you move this code into a dedicated function (something like `unmap_regions()`), that you call from here.

Reply:
Done.
Introduce a new method.